store/sqlstore/store.go

```diff
@@ -1167,6 +1172,24 @@ func (ss *SqlStore) ensureMinimumDBVersion(ver string) (bool, error) {
 	return true, nil
 }
 
+func (ss *SqlStore) ensureDatabaseCollation() error {
+	if *ss.settings.DriverName == model.DatabaseDriverMysql {
+		var dbCollation struct {
+			Variable_name string
+			Value         string
+		}
+		if err := ss.GetMasterX().Get(&dbCollation, "SHOW VARIABLES LIKE 'collation_database'"); err != nil {
+			return errors.Wrap(err, "unable to select variables")
+		}
+
+		if dbCollation.Value != "utf8mb4_general_ci" {
```

Comment:
Rather than an allowlist, can we use a blocklist of collations which are known to not work? Then we can be sure and just return an error. I think there are lots of working databases out there with a collation not equal to `utf8mb4_general_ci`. Showing a warning message every time the server starts might cause some grumpiness.